### PR TITLE
Removal of redundant normalisation of weights

### PR DESCRIPTION
**Pull request by BelligerG**
Removal of a few redundant functions. Where weighted means are requested using Iris collapsed, the weights are normalised within numpy.ma.average (which is called by iris), as such, we can remove any normalisation of weights/checking of normalisation of weights.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ improver_tests/blending/spatial_weights/test_SpatiallyVaryingWeightsFromMask.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5542849&bid=18172866)

**Comment by fionaRust** [comment]
@BelligerG As with Tomek's neighbourhood processing refactor that removed similar renormalisation functionality, I wouldn't mind reviewing this, especially to take a look at the unit test coverage. Unfortunately I may not get chance until tomorrow.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ improver_tests/blending/spatial_weights/test_SpatiallyVaryingWeightsFromMask.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5542849&bid=18172866)

**Comment by BelligerG** [comment]
> The changes to remove the `normalised_masked_weights` function, usage of it and associated tests all look OK.
> 
> With the remaining unit tests, there are changes to expected results, but it's not clear to me why the new expected results are correct - the change between before and after varies with each non-zero entry. Within a particular test, it seems to be either an overall increase or an overall decrease.
> Could you explain these changes? It may be better to do that in the PR description above or a reply comment rather than putting comments inside the code.
> 
> I've run the acceptance test suite and it all passes as-is with no output changes - that makes me suspect that these unit tests which need updating are overly-coupled to internal details.
> 
> There's also a long commit history similar to #1260 - see my comment in that PR for suggestions on how to clean it up.

Thanks for your comment and your help yesterday, I've reduced the number of commits. The non-zero entries of the test were indeed coupled to internal details - we had generated non-normalised weights, the tests were based on normalising them and checking them, however, when we run them through the "next step" the weights are normalised anyway. So the acceptance test is testing based on that next step and passing okay.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ improver_tests/blending/spatial_weights/test_SpatiallyVaryingWeightsFromMask.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5542849&bid=18172866)

**Comment by tjtg** [comment]
> The non-zero entries of the test were indeed coupled to internal details - we had generated non-normalised weights, the tests were based on normalising them and checking them, however, when we run them through the "next step" the weights are normalised anyway. So the acceptance test is testing based on that next step and passing okay.

I'm happy with that explanation of why the unit test expected numbers have changed. I'll mark this as approved once @fionaRust has done her review.

File: improver/blending/spatial_weights.py [hunk 1]
```diff
@@ -216,51 +216,6 @@ def multiply_weights(weights_from_mask, one_dimensional_weights_cube, blend_coor
         result = result.merge_cube()
         return result
 
-    @staticmethod
-    def normalised_masked_weights(weights_cube, blend_coord):
```
Comment (fionaRust) [inline improver/blending/spatial_weights.py:220]: There are some docstrings that now need updating to say the weights returned from this plugin aren't normalised

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ improver_tests/blending/spatial_weights/test_SpatiallyVaryingWeightsFromMask.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5542849&bid=18172866)

File: improver/blending/spatial_weights.py [hunk 2]
```diff
@@ -317,7 +317,7 @@ def process(self, cube_to_collapse, one_dimensional_weights_cube, blend_coord):
 
         Returns:
             iris.cube.Cube:
-                A cube containing normalised spatial weights based on the
+                A cube containing non-normalised spatial weights based on the
```
Comment (fionaRust) [inline improver/blending/spatial_weights.py:320]: Sorry I think you need to check the class docstring too.

Comment (BelligerG) [reply improver/blending/spatial_weights.py:320]: Ah, apologies, I see it, I'll fix it now

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ improver_tests/blending/spatial_weights/test_SpatiallyVaryingWeightsFromMask.py  -8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5542849&bid=18172866)

**Pull request merged**